mentat/conversation.py

```diff
@@ -2,50 +2,46 @@
 
 import json
 import logging
-from enum import Enum
 from timeit import default_timer
 from typing import TYPE_CHECKING
 
-from openai.error import RateLimitError
+from openai import RateLimitError
+from openai.types.chat import ChatCompletionMessageParam
 
 from mentat.errors import MentatError
-from mentat.llm_api import (
-    call_llm_api,
+from mentat.llm_api_handler import (
     conversation_tokens,
     count_tokens,
     get_prompt_token_count,
-    is_model_available,
     model_context_size,
 )
 from mentat.session_context import SESSION_CONTEXT
+from mentat.utils import add_newline
 
 if TYPE_CHECKING:
     from mentat.parsers.file_edit import FileEdit
 
 
-class MessageRole(Enum):
-    System = "system"
-    User = "user"
-    Assistant = "assistant"
```

Comment:
why remove this Enum? Seems cleaner to me than just using the strings everywhere

Reply:
I did this because OpenAI's typing uses Literal['user', 'system', 'assistant']; while we can go back to Enums, I think we should stick with OpenAI's typing (we're just going to be role.value everywhere with the Enum, and of course pyright will catch any errors because it is a Literal)

Reply:
I see, didn't realize it was Literal['user', 'system', 'assistant'] and pyright would still catch. I'm ok with it then